builtin/providers/cloudstack/resource_cloudstack_private_gateway.go

```diff
@@ -0,0 +1,173 @@
+package cloudstack
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/xanzy/go-cloudstack/cloudstack"
+)
+
+func resourceCloudStackPrivateGateway() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudStackPrivateGatewayCreate,
+		Read:   resourceCloudStackPrivateGatewayRead,
+		Update: resourceCloudStackPrivateGatewayUpdate,
+		Delete: resourceCloudStackPrivateGatewayDelete,
+
+		Schema: map[string]*schema.Schema{
+			"gateway": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"ip_address": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"netmask": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"vlan": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"physical_network_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"network_offering": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"acl_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
```

Comment:
Thinking about this again, shouldn't this be required? As you cannot first add an ACL and then later remove it again. That would cause the resource to break right? So if we just require this field, we are sure the resource will never break. Thoughts?